### PR TITLE
Add aggregable badges to response

### PR DESCRIPTION
**Pull request by efritz**
Sister PR: https://github.com/sourcegraph/sourcegraph/pull/18843.

**TODO**:

- [x] remove casts to any after we can upgrade sourcegraph type def package

### CODE REVIEW

File: shared/providers.ts [hunk 1]
```diff
@@ -286,8 +290,15 @@ export function createReferencesProvider(
...
-            for await (const lsifResult of lsifProvider(textDocument, position, context)) {
-                if (nonEmpty(lsifResult)) {
+            for await (const rawResult of lsifProvider(textDocument, position, context)) {
+                if (nonEmpty(rawResult)) {
+                    const lsifResult = asArray(rawResult).map(
+                        result =>
+                            (({
+                                ...result,
+                                aggregableTags: [{ text: 'semantic', linkURL: HoverAlerts.linkURL }],
+                            } as any) as sourcegraph.Location)
```
Comment (Strum355) [inline shared/providers.ts:300]: are both of these needed? :thinking:

Comment (efritz) [reply shared/providers.ts:300]: Yes right now this is v. v. stupid (there's no type intersection so you have to upcast to top before downcasting to sourcegraph.Location).

Once we publish the type def in the sister PR we won't need any casting.

**Review by Strum355 — APPROVED** [review]
Pending my comment on the sister PR :heavy_check_mark:

File: shared/indicators.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    linkURL,
+    hoverMessage:
+        'It looks like this symbol is defined in another repository that does not have a pre-computed semantic index. Go to definition may be imprecise.',
+}
+
+export const partialDefinitionNoHoverBadge: AggregableBadge = {
+    text: 'partial semantic',
+    linkURL,
+    hoverMessage:
+        'It looks like this symbol is defined in another repository that does not have a pre-computed semantic index. This hover text may be imprecise.',
```
Comment (shrouxm) [inline shared/indicators.ts:42]: not sure if the right place for convo about this text is elsewhere, but 
`s/a pre-computed semantic index/semantic indexing/g`?

Comment (efritz) [reply shared/indicators.ts:42]: This was your text! We can change it to whatever we like:

- It looks like this symbol is defined in another repository that does not have a pre-computed semantic index.
- It looks like this symbol is defined in another repository that does not have semantic indexing.
- It looks like this symbol is defined in another repository that does not have semantic indexing enabled.
- It looks like this symbol is defined in another repository that is not being semantically indexed.

Comment (efritz) [reply shared/indicators.ts:42]: cc @AlicjaSuska

**Review by Strum355 — APPROVED** [review]

**Pull request merged**